scala-generator/src/main/scala/models/generator/ScalaCaseClasses.scala

```diff
@@ -98,9 +98,15 @@ trait ScalaCaseClasses extends CodeGenerator {
   }
 
   def generateCaseClass(model: ScalaModel, unions: Seq[ScalaUnion]): String = {
+    val extendsAnyval = if(model.isValueClass) Some(" extends AnyVal") else None
+
+    val extendsClause = ScalaUtil.extendsClause(unions.map(_.name))
```

Comment:
Note: the typical extends clause has priority over `AnyVal` on purpose - most of the time it wouldn't give us any profit to wrap something that's part of an ADT, and I'm not even sure it'd compile.